### DISCUSSION TITLE
calling one use case from another use case

### DISCUSSION THREAD

**Discussion by a28028** [jasontaylordev/CleanArchitecture#1200] [General]
hi
In a Clean Architecture approach using C#, calling one use case from another is a common scenario. Let's say we have two use cases: UseCaseA and UseCaseB, where UseCaseA needs to call UseCaseB.

For example, the CreateInvoiceCommand needs to call the CreateWalletCommandHandler.

How should this be implemented in a Clean Architecture framework?

**Comment by DennisJansenDev** [comment]
Through domain events and domain event handlers. Never directly call another command from a command handler.

See: https://medium.com/workleap/why-command-should-not-call-command-in-cqrs-5da046a9fed1

**Comment by a28028** [comment]
Could you provide an example of a possible solution for the following scenario: To avoid such problems, it is generally recommended to keep commands self-contained and not have them call other commands directly. Instead, if there is a need for multiple commands to be executed together, a higher-level command or workflow can be created to orchestrate the execution of the individual commands.

**Comment by a28028** [comment]
hi 
My issue was resolved with the solution provided at https://www.jamesmichaelhickey.com/domain-driven-design-aggregates/
